zerver/actions/user_groups.py

```diff
@@ -472,8 +479,36 @@ def add_subgroups_to_user_group(
     ]
     RealmAuditLog.objects.bulk_create(audit_log_entries)
 
+    subscriber_ids_for_streams = get_user_ids_for_streams({stream.id for stream in streams})
+    new_stream_metadata_user_ids = bulk_can_access_stream_metadata_user_ids(streams)
+    recent_traffic = get_streams_traffic({stream.id for stream in streams}, realm)
+    anonymous_group_membership = get_anonymous_group_membership_dict_for_streams(streams)
+
     do_send_subgroups_update_event("add_subgroups", user_group, subgroup_ids)
 
+    for stream in streams:
+        user_ids_gaining_metadata_access = (
+            new_stream_metadata_user_ids[stream.id] - old_stream_metadata_user_ids[stream.id]
+        )
+        send_stream_creation_event(
+            realm,
+            stream,
+            list(user_ids_gaining_metadata_access),
+            recent_traffic,
+            anonymous_group_membership,
+        )
+        peer_add_event = dict(
+            type="subscription",
+            op="peer_add",
+            stream_ids=[stream.id],
+            user_ids=sorted(subscriber_ids_for_streams[stream.id]),
+        )
+        send_event_on_commit(
+            realm,
+            peer_add_event,
+            user_ids_gaining_metadata_access,
+        )
+
```

Comment:
Not sure if we should de-duplicate this code, seems simple enough code to just let it be